src/Agent.Plugins/GitSourceProvider.cs

```diff
@@ -1219,6 +1219,11 @@ public async Task PostJobCleanupAsync(AgentTaskPluginExecutionContext executionC
             }
         }
 
+        protected virtual GitCliManager GetCliManager(Dictionary<string, string> gitEnv = null)
```

Comment:
Would it be better to just constructor inject this intead of making a virtual method to change the implementation? Not sure if we know `gitEnv` at creation time.